test/utils/test-utils.js

```diff
@@ -178,4 +178,18 @@ function copyFile(testCase, file) {
     }
 }
 
-module.exports = { run, runWatch, runAndGetWatchProc, extractSummary, appendDataIfFileExists, copyFile, copyFileAsync, appendDataToMultipleIfFilesExists };
+/**
+ * It splits the arr passed as an argument using node.js's seperator for supporting cross-windows
+ * matching for the paths, and slice the elements from the idx passed and
+ * joins them to create a string which can be matched against the output directory
+ * @param {Array} arr - the array element, in this case the outputDir array from summary
+ * @param {Number} idx - index from the end to start slicing
+ * @returns {string} - output directory to test against
+ */
+function outDirExtractor(arr, idx) {
```

Comment:
Is there a way to find the idx from the array? I think so, and it would make the abstraction better

Reply:
Actually idx is kind of depenedent on the `outdir` where it needs to match against. 
Like here https://github.com/webpack/webpack-cli/pull/1074/files#diff-b82e1bc49af5272a0d78bc41925fb337R10 , idx needs to be 3 as there are 3 sub paths it's checking, so either that string need to be passed and then checking the len after slitting it with `\`,  so either way we need 2nd argument to pass  I am ready to change if you want that string to be passed instead to index

Reply:
cc @evenstensberg

Reply:
If you iterate the array and try to concat those? Maybe even check if one those three paths are the correct one, if not present, fail the test?

Reply:
> If you iterate the array and try to concat those?

I didn't get this !?

the `arr` array will be joined to form a string which will then test against the `outDir` ! so we need both right?

Did you mean removing the `outDir` and just testing the `arr` (which is the output came from the summary) by checking the file system directories? if so, wouldn't it be a lot of work which can be reduced by doing manually?

Reply:
You'll always have a knowing of these output directories, so it's not always needed to provide an array.

Reply:
@evenstensberg I didn't refactore this as I am still not clear about the approach. 
Should I remove this function and also those assertions where this is being used.